Mathlib/Data/FunLike/Basic.lean

```diff
@@ -221,4 +221,16 @@ protected theorem congr_arg (f : F) {x y : α} (h₂ : x = y) : f x = f y :=
 
 end FunLike
 
+/-- The class `NDFunLike F α β` expresses that terms of type `F` have an
+injective coercion to functions from `α` to `β`.
+
+This typeclass is used in the definition of the homomorphism typeclasses,
+such as `ZeroHomClass`, `MulHomClass`, `MonoidHomClass`, ....
+-/
+class NDFunLike (F : Sort*) (α : outParam <| Sort*) (β : outParam <| Sort*) extends
+  FunLike F α fun _ ↦ β
+
+instance (priority := 110) hasCoeToFun {F α β} [NDFunLike F α β] : CoeFun F fun _ ↦ α → β where
+  coe := NDFunLike.toFunLike.coe
```

Comment:
Oh, I take back my other comment; this actually does continue to use FunLike.coe` as a head symbol!